### PR TITLE
BCF-3276: refactor migrations to use goose provider

### PR DESCRIPTION
**Pull request by krehermann**
[BCF-3276](https://smartcontract-it.atlassian.net/browse/BCF-3276)

Update goose and refactor migrations to use the newly added `goose.Provider`

This is a concrete step toward supporting multiple migrations providers.

[BCF-3276]: https://smartcontract-it.atlassian.net/browse/BCF-3276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/store/migrate/migrations/0035_create_feeds_managers.sql [hunk 1]
```diff
@@ -10,4 +10,4 @@ CREATE TABLE feeds_managers (
     updated_at timestamp with time zone NOT NULL
 );
 -- +goose Down
-	DROP TABLE feeds_managers
+	DROP TABLE feeds_managers;
```
Comment (krehermann) [inline core/store/migrate/migrations/0035_create_feeds_managers.sql:13]: fix a linting issue found by the new version of goose

File: core/store/migrate/migrate.go [hunk 2]
```diff
@@ -25,20 +27,55 @@ var embedMigrations embed.FS
...
 	verbose, _ := strconv.ParseBool(logMigrations)
-	goose.SetVerbose(verbose)
+
+	fys, err := fs.Sub(embedMigrations, MIGRATIONS_DIR)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sub filesystem for embedded migration dir: %w", err)
+	}
+	// hack to work around global go migrations
+	// https: //github.com/pressly/goose/issues/782
+	goose.ResetGlobalMigrations()
```
Comment (vyzaldysanchez) [inline core/store/migrate/migrate.go:52]: any concerns of the usage of this hack?

**Review by vyzaldysanchez — DISMISSED** [review]

**Comment by vyzaldysanchez** [comment]
Needs a changeset though... Will this be considered a minor change?

File: core/store/migrate/migrate.go (hunk 2 above)
Comment (krehermann) [reply core/store/migrate/migrate.go:52]: no, goose has already fixed the issue and it will be in a new release soon

**Review by vyzaldysanchez — DISMISSED** [review]

**Review by jmank88 — DISMISSED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**